app/main.py

```diff
@@ -94,26 +94,8 @@ def __eq__(self, other: int | float) -> bool | TypeError:
             f"But received {type(other)}"
         )
 
-    def __le__(self, other: int | float) -> bool | TypeError:
-        if isinstance(other, Distance):
-            return self.km <= other.km
-        if isinstance(other, (int, float)):
-            other = Distance(other)
-            return self.km <= other.km
-        raise TypeError(
-            f"unsupported type for __le__ method. <class 'float'> "
-            f"or <class 'int'> were expected. "
-            f"But received {type(other)}"
-        )
+    def __le__(self, other: int | float | Distance) -> bool | TypeError:
+        return self.__eq__(other) or self.__lt__(other)
```

Comment:
but in this case won't be able to raise a custom error for ```__le__``` and ```__ge__```, it will raise an error from ```__eq__```,```__lt__ or ```__gt__```